backend/test_observer/controllers/test_executions/logic.py

```diff
@@ -49,3 +58,124 @@ def parse_c3_test_result_status(status: C3TestResultStatus) -> TestResultStatus:
             return TestResultStatus.FAILED
         case C3TestResultStatus.SKIP:
             return TestResultStatus.SKIPPED
+
+
+def get_historic_artefact_builds_query(
+    session: Session,
+    artefact: Artefact,
+    architecture: str,
+) -> RowReturningQuery:
+    """
+    Helper method to get a query that fetches the latest Artefact Build IDs
```

Comment:
latest or historic?

Reply:
Not sure what is a better naming option here. Maybe I should rename all methods/docstrings to previous, since they are previous to the current test execution.

Reply:
previous also makes sense. But make sure to update the DTO object too